sql/src/main/java/io/crate/analyze/InsertFromValuesAnalyzer.java

```diff
@@ -330,4 +344,48 @@ private Object processPartitionedByValues(final ColumnIdent columnIdent, Object
         }
         return null;
     }
+
+    private Object[] processGeneratedExpressions(DocTableRelation tableRelation,
+                                             ExpressionAnalysisContext expressionAnalysisContext,
+                                             ExpressionReferenceValueAnalyzer referenceValueAnalyzer,
+                                             InsertFromValuesAnalyzedStatement context,
+                                             Object[] insertValues) {
+        referenceValueAnalyzer.values(insertValues);
```

Comment:
matter of taste, but:
couldn't you also create a new ExpressionReferenceValueAnalyzer for each call of this function?
mutable state, you know?